File: Algorithms/1417.Reformat-The-String/reformat_the_string.go
import "strings"

func reformat(s string) string {
	var str_arr []string
	var digit_arr []string
	for _, ele := range s {
		if 48 <= ele && ele <= 58 {
			digit_arr = append(digit_arr, string(ele))
		} else {
			str_arr = append(str_arr, string(ele))
		}
	}
	lend := len(digit_arr)
	lens := len(str_arr)
	len_diff := lend - lens
	if len_diff > 1 || len_diff < -1 {
		return ""
	}
	res := make([]string, lend+lens)
	pos := 0
	if lend > lens {
		pos = 0
	} else {
		pos = 1
	}
	for _, ele := range digit_arr {
		res[pos] = ele
		pos += 2
	}
	if lend > lens {
		pos = 1
	} else {
		pos = 0
	}
	for _, ele := range str_arr {
		res[pos] = ele
		pos += 2
	}
	return strings.Join(res, "")
}

File: Algorithms/1417.Reformat-The-String/reformat_the_string.py
class Solution:
    def reformat(self, s: str) -> str:
        length = len(s)
        arr = ['' for i in range(length + 1)]
        digit_pos = 1
        str_pos = 2
        for ele in s:
            if str.isdigit(ele):
                if digit_pos > length:
                    return ""
                arr[digit_pos] = ele
                digit_pos += 2
            else:
                if str_pos == length + 1:
                    arr[0] = ele
                    str_pos += 2
                    continue
                if str_pos > length + 1:
                    return ""
                arr[str_pos] = ele
                str_pos += 2
        return "".join(arr)
        
